src/components/Map/MapInteractionPane.tsx

```diff
@@ -1,26 +1,42 @@
 import React from "react";
+
+import { useSelector, useDispatch } from "react-redux";
+import { deviceSliceActions } from "@features/device/deviceSlice";
+import { selectIsNewWaypoint } from "@features/device/deviceSelectors";
+
 import {
-  // MapIconButton,
+  MapIconButton,
   MapIconUnimplemented,
 } from "@components/Map/MapIconButton";
 import { MapPinIcon, Square3Stack3DIcon } from "@heroicons/react/24/outline";
 
-// Currently implemented using MapIconUnimplemented, which allows for a popup.
-// When functionality is implemented, change component to MapIconButton.
-
 const MapInteractionPane = () => {
+  const dispatch = useDispatch();
+
+  const newWaypoint = useSelector(selectIsNewWaypoint());
+
+  // Toggles newWaypoint state, which allows for creation of new waypoints on map
+  const handleClickMapPin = () => {
+    dispatch(deviceSliceActions.setNewWaypoint(!newWaypoint));
```

Comment:
Not sure if the naming is clear here, what does the `newWaypoint` state represent?